src/mainsection.jsx

```diff
@@ -12,29 +15,53 @@ export default function MainSection() {
             }
             return newLife;
         });
+
+        const newHiddenSquares = [...hiddenSquares];
+        for (let i = 0; i < 4; i++) {
+            let randomIndex;
+            do {
```

Comment:
This can be refactored to:

```
        for (let i = 0; i < 4; i++) {
            newHiddenSquares.push(Math.floor(Math.random() * 24));
        }
```